Digital Credentials: drop navigator.identity, use navigator.credentials instead

marcoscaceres opened the pull request:
#### d1443bb6e59d2e4b597aa3649c35eb3637a165f6
<pre>
Digital Credentials: drop navigator.identity, use navigator.credentials instead
<a href="https://bugs.webkit.org/show_bug.cgi?id=279159">https://bugs.webkit.org/show_bug.cgi?id=279159</a>
<a href="https://rdar.apple.com/135835222">rdar://135835222</a>

Reviewed by Abrar Rahman Protyasha.

Drop the navigator.identity namespace, and do everything through navigator.credentials instead.

Fix Credentials Container&apos;s create method to resolve a null, per spec default create algorithm:
<a href="https://www.w3.org/TR/credential-management-1/#algorithm-create-cred">https://www.w3.org/TR/credential-management-1/#algorithm-create-cred</a>

Spec change:
<a href="https://github.com/WICG/digital-credentials/pull/164">https://github.com/WICG/digital-credentials/pull/164</a>

Web platform Tests Upstream commit:
<a href="https://github.com/web-platform-tests/wpt/commit/fd7b0de1938f9ed8abe3974803ddd2196c5880f0">https://github.com/web-platform-tests/wpt/commit/fd7b0de1938f9ed8abe3974803ddd2196c5880f0</a>

* LayoutTests/http/wpt/identity/identitycredentialscontainer-create-basics.https-expected.txt: Removed.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-create-basics.https.html: Removed.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-basics.https-expected.txt:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-basics.https.html:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-hidden.https-expected.txt:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-hidden.https.html:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-store-basics.https-expected.txt: Removed.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-store-basics.https.html: Removed.
* LayoutTests/http/wpt/identity/idl.https-expected.txt:
* LayoutTests/http/wpt/identity/idl.https.html:
* LayoutTests/http/wpt/identity/setDigitalCredentialsEnable.https.html:
* LayoutTests/imported/w3c/web-platform-tests/credential-management/credentialscontainer-create-basics.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/credential-management/credentialscontainer-create-basics.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/create.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/create.tentative.https.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https.html.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/dc-types.ts:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/default-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/enabled-on-self-origin-by-permissions-policy.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/enabled-on-self-origin-by-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get.tentative.https.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https.html.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.js: Added.
(test):
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/non-fully-active.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/helper.js:
(export.makeGetOptions):
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/iframe.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/permissions-policy/resources/digital-credentials-get.html:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/get.tentative.https-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Removed.
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk2/fast/dom/navigator-detached-no-crash-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
(WebCore::CredentialsContainer::get):
(WebCore::CredentialsContainer::isCreate):
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.h:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp:
(WebCore::CredentialRequestCoordinator::discoverFromExternalSource):
* Source/WebCore/Modules/identity/IdentityCredentialsContainer.cpp:
(WebCore::IdentityCredentialsContainer::get):
* Source/WebCore/Modules/identity/Navigator+Identity.idl: Removed.
* Source/WebCore/Modules/identity/NavigatorIdentity.cpp: Removed.
* Source/WebCore/Modules/identity/NavigatorIdentity.h: Removed.
* Source/WebCore/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: <a href="https://commits.webkit.org/289533@main">https://commits.webkit.org/289533@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/68cb413d5f96108cc31abc1cb40fb82fb62110b3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87094 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6604 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41452 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91953 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37833 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89144 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6880 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14672 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67313 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25063 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90096 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5264 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78831 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47635 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5040 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33207 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36950 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75531 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34086 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93840 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14256 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10374 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76122 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-type-on-non-root.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14460 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74687 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75321 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18559 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19658 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18092 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7173 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14275 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19568 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14020 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17462 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15801 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/DerivedSources.make:
```diff
@@ -384,7 +384,6 @@ JS_BINDING_IDLS := \
     $(WebCore)/Modules/identity/DigitalCredentialRequestOptions.idl \
     $(WebCore)/Modules/identity/OpenID4VPRequest.idl \
     $(WebCore)/Modules/identity/IdentityCredentialProtocol.idl \
-    $(WebCore)/Modules/identity/Navigator+Identity.idl \
```
You'll also have to remove the file reference from `Source/WebCore/CMakeLists.txt` to unblock the GTK build.

aprotyas reviewed (commented):
I don't have the full picture on the test changes, but the engine changes are LGTM.

aprotyas reviewed (approved):
`gtk-wk2` failure looks unrelated.

marcoscaceres committed:
Digital Credentials: drop navigator.identity, use navigator.credentials instead
https://bugs.webkit.org/show_bug.cgi?id=279159
rdar://135835222

Reviewed by Abrar Rahman Protyasha.

Drop the navigator.identity namespace, and do everything through navigator.credentials instead.

Fix Credentials Container's create method to resolve a null, per spec default create algorithm:
https://www.w3.org/TR/credential-management-1/#algorithm-create-cred

Spec change:
https://github.com/WICG/digital-credentials/pull/164

Web platform Tests Upstream commit:
https://github.com/web-platform-tests/wpt/commit/fd7b0de1938f9ed8abe3974803ddd2196c5880f0

* LayoutTests/http/wpt/identity/identitycredentialscontainer-create-basics.https-expected.txt: Removed.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-create-basics.https.html: Removed.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-basics.https-expected.txt:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-basics.https.html:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-hidden.https-expected.txt:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-hidden.https.html:
* LayoutTests/http/wpt/identity/identitycredentialscontainer-store-basics.https-expected.txt: Removed.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-store-basics.https.html: Removed.
* LayoutTests/http/wpt/identity/idl.https-expected.txt:
* LayoutTests/http/wpt/identity/idl.https.html:
* LayoutTests/http/wpt/identity/setDigitalCredentialsEnable.https.html:
* LayoutTests/imported/w3c/web-platform-tests/credential-management/credentialscontainer-create-basics.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/credential-management/credentialscontainer-create-basics.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/allow-attribute.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/create.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/create.tentative.https.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https.html.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/dc-types.ts:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/default-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/enabled-on-self-origin-by-permissions-policy.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/enabled-on-self-origin-by-permissions-policy.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get-user-activation.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get.tentative.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/get.tentative.https.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https.html.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.js: Added.
(test):
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/historical.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-create.tentative.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/non-fully-active.https.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/helper.js:
(export.makeGetOptions):
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/support/iframe.html:
* LayoutTests/imported/w3c/web-platform-tests/digital-credentials/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/permissions-policy/resources/digital-credentials-get.html:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/disabled-by-permissions-policy.https.sub-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/get.tentative.https-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/digital-credentials/identity-get.tentative.https-expected.txt: Removed.
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk2/fast/dom/navigator-detached-no-crash-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
(WebCore::CredentialsContainer::get):
(WebCore::CredentialsContainer::isCreate):
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.h:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp:
(WebCore::CredentialRequestCoordinator::discoverFromExternalSource):
* Source/WebCore/Modules/identity/IdentityCredentialsContainer.cpp:
(WebCore::IdentityCredentialsContainer::get):
* Source/WebCore/Modules/identity/Navigator+Identity.idl: Removed.
* Source/WebCore/Modules/identity/NavigatorIdentity.cpp: Removed.
* Source/WebCore/Modules/identity/NavigatorIdentity.h: Removed.
* Source/WebCore/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: https://commits.webkit.org/289533@main
